### ISSUE TITLE
Update documentation and guides related to WinForms Accessibility

### ISSUE THREAD

**Issue by M-Lipin** [dotnet/winforms#2884]
** .NET Core Version:**

.NET Core SDK (reflecting any global.json):
 Version:   5.0.100-alpha.1.20073.10
 Commit:    29f4d693a9

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.18363
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\5.0.100-alpha1-05536\

**Have you experienced this same bug with .NET Framework?:**
Yes

**Problem description:**
WinForms documentation and guides do not fully describe the steps and options about setting accessibility properties:
* https://github.com/dotnet/docs/blob/live/docs/framework/winforms/controls/providing-accessibility-information-for-controls-on-a-windows-form.md
* https://docs.microsoft.com/en-us/dotnet/api/system.windows.forms.control.accessiblename?view=netframework-4.8
* https://docs.microsoft.com/en-us/dotnet/api/system.windows.forms.control.accessibledefaultactiondescription?view=netframework-4.8

**Expected behavior:**
Documentation and guides should be descriptive to allow app developers to implement accessible applications using WinForms controls.

**Minimal repro:**
Create WinForms app, add several WinForms controls to the form, try to get information how to set up accessibility properties to make the application fully accessible.
There is no default information except the `AccessibleName` has default value 'null'.

**Comment by M-Lipin** [comment]
It would be good to update the documentation with more descriptive text:

**Proposition for 'AccessibleName' property doc update:**
'AccessibleName' property is used as a control name that can be easily accessed by users with disabilities who are assisted by accessibility technologies (UI Automation)/applications to translate the form content as an accessible content (for instance audio announcement for text labels). 'AccessibleName' should have descriptive name and in most cases should be same as the control name. For instance the 'TextBox' with name _"First name"_ should have 'AccessibleName' _"First name"_ so Narrator or another accessibility app can announce _"First name TextBox"_ when navigating to it.

**Comment by weltkante** [comment]
I don't like this. Control names are not suitable to exposing via accessibility.

The designer defaults control names to variable names and often you have syntax like underscore or hungarian notation `txtUserName` where you prefix controls with an abbreviation of their type. You cannot have spaces in control names either (because variable names cannot have spaces) and you cannot localize control names (because they are used as keys into a dictionary).

Please keep the concept of control names and accessibility names separated, they serve different purposes and follow different semantics.

**Comment by Tanya-Solyanik** [comment]
Based on the following example:
> For instance the 'TextBox' with name "First name" should have 'AccessibleName' "First name" 

I think Mikhail meant `display name` , not the `Control.Name` property.

**Comment by weltkante** [comment]
Possibly, but for documentation this needs to be not ambigous, "control name" is a term most developers connect to the `Control.Name` property.

**Comment by M-Lipin** [comment]
Yes, thanks, I meant display name or the name which usually provided in the label just before the control.

**Comment by RussKie** [comment]
This issue should really moved to the docs repo.

> **Proposition for 'AccessibleName' property doc update:**
> 'AccessibleName' property is used as a control name that can be easily accessed by users with disabilities who are assisted by accessibility technologies (UI Automation)/applications to translate the form content as an accessible content (for instance audio announcement for text labels). 'AccessibleName' should have descriptive name and in most cases should be same as the control name. For instance the 'TextBox' with name _"First name"_ should have 'AccessibleName' _"First name"_ so Narrator or another accessibility app can announce _"First name TextBox"_ when navigating to it.

I wouldn't mention "disabilities" at all.
```diff
- 'AccessibleName' property is used as a control name that can be easily accessed by users with disabilities who are assisted by accessibility technologies...
+ 'AccessibleName' property is used as a control name that can be easily accessed by users who are assisted by accessibility technologies...
```

**Comment by merriemcgaw** [comment]
@dmitrii-drobotov can you confirm if there is work for us here?

**Comment by dmitrii-drobotov** [comment]
> @dmitrii-drobotov can you confirm if there is work for us here?

Sure, will take a look.
